src/test/java/io/keycloak/builder/Utils.java

```diff
@@ -2,17 +2,58 @@
 
 import static io.quarkus.bootstrap.util.PropertyUtils.isWindows;
 
-import java.io.File;
-import java.io.IOException;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+
 import org.apache.commons.io.FileUtils;
 
 public class Utils {
 
+    String SCRIPT_CMD = isWindows() ? "kc.bat" : "kc.sh";
+    String SCRIPT_CMD_INVOKABLE = isWindows() ? SCRIPT_CMD : "./" + SCRIPT_CMD;
+    static final String EXPECTED_OUTPUT_FROM_SUCCESSFULLY_STARTED = "Installed features";
+    static final String EXPECTED_OUTPUT_FROM_FAILURE_STARTED = "For more details";
+
+    public boolean isApplicationStaredOrFailed(Process process) throws IOException {
+        boolean isApplicationFinished = false;
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8))) {
+            String line;
+            StringBuilder stringBuilder = new StringBuilder();
+            while ((line = in.readLine()) != null) {
+                System.out.println(line);
+                stringBuilder.append(line);
+                stringBuilder.append("/n");
+            }
+            if (stringBuilder.toString().contains(EXPECTED_OUTPUT_FROM_SUCCESSFULLY_STARTED) || stringBuilder.toString().contains(EXPECTED_OUTPUT_FROM_FAILURE_STARTED)) {
```

Comment:
@aloubyansky: When I have stored the console logs in StringBuilder to ensure that Output Buffer is empty, It never comes to this point. It keeps reading the logs. Maybe (line = in.readLine()) is never null on the terminal)

Reply:
Right. It's because `in.readLine()` blocks waiting for new output from the process. It will return `null` only when the process has terminated, which in this case happens when you call `process.destroy()`.
So in this case, we have two options:
1. keep it the way you had it before, i.e. look for the line indicating a successful start and then destroy the process. The output buffer of the Keycloak process won't be completely empty, afaics, but probably not full enough to cause a hang.
2. keep it the way you had it before but move read the rest of the output buffer after you called `processKC.destroy()` to make sure there is nothing left in the buffer.